src/main/java/com/nyxei/pollingus/WhatsappApiServiceGenerator.java

```diff
@@ -0,0 +1,85 @@
+package com.nyxei.pollingus;
+
+import com.nyxei.pollingus.configuration.WhatsappApiConfig;
+
+//import com.nyxei.pollingus.exception.WhatsappApiException;
+import com.nyxei.pollingus.interceptor.AuthenticationInterceptor;
+import okhttp3.OkHttpClient;
+import java.io.IOException;
+import retrofit2.Converter;
+import retrofit2.Response;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public class WhatsappApiServiceGenerator {
+
+    private static final OkHttpClient sharedClient;
+    private static final Converter.Factory converterFactory = JacksonConverterFactory.create();
+
+    static {
+
+        sharedClient = new OkHttpClient.Builder()//
+                .callTimeout(20, TimeUnit.SECONDS)//
+                .pingInterval(20, TimeUnit.SECONDS)//
+                .build();
+    }
+
+   
+
+
+
+    public static <S> S createService(Class<S> serviceClass, String token, String baseUrl) {
+        Retrofit.Builder retrofitBuilder = new Retrofit.Builder().baseUrl(baseUrl).addConverterFactory(converterFactory);
+
+        if (token == null) {
+            retrofitBuilder.client(sharedClient);
```

Comment:
If the purpose of `Retrofit.Builder.client()` method is to set a client, you should rename the method.